decoder.go

```diff
@@ -20,18 +32,45 @@ func frameLineSize(frame *C.AVFrame) *C.int {
 	return (*C.int)(unsafe.Pointer(&frame.linesize[0]))
 }
 
-// h264Decoder is a wrapper around ffmpeg's H264 decoder.
-type h264Decoder struct {
-	codecCtx    *C.AVCodecContext
-	srcFrame    *C.AVFrame
-	swsCtx      *C.struct_SwsContext
-	dstFrame    *C.AVFrame
-	dstFramePtr []uint8
+// getStreamInfo opens a stream URL and retrieves its information
+func getStreamInfo(url string) string {
+	// Register all formats and codecs
+	C.avformat_network_init()
+
+	// Convert Go string to C string
+	cUrl := C.CString(url)
+	defer C.free(unsafe.Pointer(cUrl))
+
+	// Allocate an AVFormatContext.
+	var avFormatCtx *C.AVFormatContext = nil
+
+	// Try to open the stream
+	if C.avformat_open_input(&avFormatCtx, cUrl, nil, nil) != 0 {
+		return "Error: Could not open the stream"
+	}
+	defer C.avformat_close_input(&avFormatCtx)
+
+	// Try to retrieve the stream information
+	if C.avformat_find_stream_info(avFormatCtx, nil) < 0 {
+		return "Error: Could not retrieve the stream information"
+	}
+
+	// Iterate over all streams and check codec ID
+	for i := C.uint(0); i < C.uint(avFormatCtx.nb_streams); i++ {
```

Comment:
Are we sure that go i++ does the right thing with C `uint`s? 
Could I recommend instead:
```
for i := 0; i < int(avFormatCtx.nb_streams); i++ {
...
}
```